src/web/app/src/components/DependenciesPage.tsx

```diff
@@ -0,0 +1,60 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { useEffect, useState } from 'react';
+
+import DependenciesTable from './DependenciesTable';
+import { dependencyDiscoveryUrl } from '../config';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    backgroundColor: theme.palette.background.default,
+    fontFamily: 'Spartan',
+    padding: '1em 0 2em 0',
+    paddingTop: 'env(safe-area-inset-top)',
+    wordWrap: 'break-word',
+    [theme.breakpoints.down(1024)]: {
+      maxWidth: 'none',
+    },
+    '& h1': {
+      color: theme.palette.text.secondary,
+      fontSize: 24,
+      transition: 'color 1s',
+      marginTop: 0,
+    },
+  },
+  container: {
+    padding: '2vh 18vw',
+    [theme.breakpoints.down(1024)]: {
+      padding: '2vh 8vw',
+      wordWrap: 'break-word',
+    },
+  },
+}));
+
+const DependenciesPage = () => {
+  const classes = useStyles();
+  const [dependencies, setDependencies] = useState<string[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const fetchDependenciesData = await fetch(`${dependencyDiscoveryUrl}/projects`);
```

Comment:
Can you switch to use `useSWR` instead, so it caches?

Reply:
I'm not too familiar with `SWR`, but what I understand, it helps to cache our Data on the first run. And if during the page render some other component call the same URL using `SWR`, It will just use the cache instead of making another call.
Correct me if I'm wrong.

Reply:
We use it throughout the front-end, it is a React Hook wrapper over `fetch` that improves its ergonomics:

```js
const { data, error } = useSWR('/api/user', fetcher)
```

It will do all sorts of things to help you, like retrying failed requests, caching the data and giving you old data while it reloads, etc.

You could also use it to do infinite scrolling on your deps list later if you wanted.